src/dropdown/dropdown.ts

```diff
@@ -101,13 +110,11 @@ export class NgbDropdownToggle extends NgbDropdownAnchor {
 @Directive({
   selector: '[ngbDropdown]',
   exportAs: 'ngbDropdown',
-  host: {
-    '[class.show]': 'isOpen()',
-    '(keyup.esc)': 'closeFromOutsideEsc()',
-    '(document:click)': 'closeFromClick($event)'
-  }
+  host: {'[class.show]': 'isOpen()', '(document:click)': 'closeFromClick($event)'}
 })
-export class NgbDropdown implements OnInit {
+export class NgbDropdown implements OnInit,
+    OnDestroy {
+  private _closed$ = new Subject();
```

Comment:
hmm, why create a new one, if you already have `openChange`?

`takeUntil(this.openChange.pipe(filter(opened => opened === false)))` ?